mri/reconstruct/fourier.py

```diff
@@ -208,3 +208,139 @@ def adj_op(self, x):
         """
         self.plan.f = x
         return np.copy(self.plan.adjoint()) / np.sqrt(self.plan.M)
+
+
+class Stacked3D:
+    """"  3-D non uniform Fast Fourrier Transform class,
+        fast implementation for Stacked samples
+    Attributes
+    ----------
+    samples: np.ndarray
+        the mask samples in the Fourier domain.
+    shape: tuple of int
+        shape of the image (necessarly a square/cubic matrix).
+    fft_type: 'NFFT' | 'NUFFT' default 'NFFT'
+        What FFT version to be used for 2D non uniform FFT
+    platform: string, 'cpu', 'multi-cpu' or 'gpu' default 'gpu'
+        string indicating which hardware platform will be used to compute the
+        NUFFT. works only if fft_type=='NUFFT'
+    """
+
+    def __init__(self, samples, shape, fft_type='NFFT', platform='gpu'):
```

Comment:
But, if we do that, it would mean that everytime we initialize a NUFFT, we need to pass the argument `platform='gpu'`, which seems inefficient to me. 
What we could possibly do is that keep platform as GPU by default for NUFFT and get rid of this parameter altogether, how does that sound?

Reply:
I am not sure about putting the GPU by default in the NUFFT because of two things:
First, the GPU version of the NUFFT does not come with the installation of the plugin.
Second, no travis tests have been made with the GPU version.
For those two reasons, I think we should keep the default setting to `'cpu'`, hence someone who wanted to use the GPU needs to specify it.

Reply:
Wait, I think here was my idea:
1) Use NFFT for CPU, 
2) Use NUFFT for GPU or Multiple-CPU

I dont understand why we would have to use NUFFT with CPU. But you do make a sound point that isnt tested, what is the best way to go with here?